nomad/server.go

```diff
@@ -1450,6 +1454,27 @@ func (s *Server) Stats() map[string]map[string]string {
 	return stats
 }
 
+// EmitRaftStats is used to export metrics about the blocked eval tracker while enabled
+func (s *Server) EmitRaftStats(period time.Duration, stopCh <-chan struct{}) {
+	for {
+		select {
+		case <-time.After(period):
+			commitIndex := s.raft.LastIndex()
+			metrics.SetGauge([]string{"raft", "commitIndex"}, float32(commitIndex))
```

Comment:
Should this be `logIndex` instead of `commitIndex`? At least the comments in https://github.com/hashicorp/raft/blob/master/state.go#L66 suggest (because of "cache") that we're getting a possibly stale index from the raft `max(log, snapshot)`. This may be fine, I'm still a little hazy on which ids from raft may move backwards.

Reply:
It looks to me like `logIndex` and `commitIndex` are distinct, and probably both helpful. The only place the commit index is returned from the raft api looks like its in the `Stats` call, https://github.com/hashicorp/raft/blob/master/api.go#L943